ClassicPress-plugins/classic-commerce
# Fix grunt and update autogenerated files

### Changes proposed in this Pull Request:

Fix the build process using `grunt` and update all (known to me) auto-generated files currently tracked as part of the git history.

### How to test the changes in this Pull Request:

1. Install the recommended version of Node.js and run `npm install` and `composer install`. Verify no warnings and no modified files tracked by `git`.
2. Run `grunt`. Verify overall success of the command, no modified files afterwards, and the only warnings should be the following:

```
Running "postcss:dist" (postcss) task
>> autoprefixer: /home/james/cp/research/classic-commerce/assets/css/admin-rtl.css:5705:7: Gradient has outdated direction syntax. New syntax is like `to left` instead of `right`.
>> autoprefixer: /home/james/cp/research/classic-commerce/assets/css/admin-rtl.css:5719:7: Gradient has outdated direction syntax. New syntax is like `to left` instead of `right`.
>> autoprefixer: /home/james/cp/research/classic-commerce/assets/scss/admin.scss:5804:4: Gradient has outdated direction syntax. New syntax is like `to left` instead of `right`.
>> autoprefixer: /home/james/cp/research/classic-commerce/assets/scss/admin.scss:5812:4: Gradient has outdated direction syntax. New syntax is like `to left` instead of `right`.
>> 29 processed stylesheets created.
>> 4 issues found.
```

3. The changes in `assets/js/admin/meta-boxes-order.min.js` and `assets/js/frontend/country-select.min.js` need to be explained/verified somehow. Hopefully these are expected (due to corresponding changes in the un-minified code in previous PRs), but I have not looked through the history to find these changes.

### Before:

`grunt` didn't work due to the following errors:

```sh
$ grunt
Running "checktextdomain:files" (checktextdomain) task

includes/class-wc-update-client.php
[L171]  esc_html__  Incorrect text domain used ("codepotent-update-manager")
[L172]  esc_html__  Incorrect text domain used ("codepotent-update-manager")
[L173]  esc_html__  Incorrect text domain used ("codepotent-update-manager")
[L326]  esc_html__  Incorrect text domain used ("codepotent-update-manager")
Warning: 4 problems Use --force to continue.

Aborted due to warnings.
```

I've ignored this file in the `checktextdomain:files` task since it is intentionally using a different textdomain for translated strings.

`grunt` also modified several files (minified JS code and the .pot file for translations). These are currently tracked in `git`, which may not be what is desired (especially for the .pot file), but while these files are still in there, they should be kept up to date with corresponding changes in the source files.

`composer install` generated the following warning because `composer.json` was updated at some point without a corresponding update to `composer.lock`:

> Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run `update` to update them.

Running `composer update` as described generated an updated version of `composer.lock` which has been included here.

### After:

The build process using `grunt` should work now, and running `grunt` should not result in any modified files tracked by `git`.

`composer install` should not result in any warnings.
